actors/verifreg/src/lib.rs

```diff
@@ -762,6 +811,133 @@ where
     )
 }
 
+// Deserializes and validates a token receiver hook payload.
+fn validate_tokens_received(
+    raw: &RawBytes,
+    my_id: u64,
+) -> Result<TokensReceivedParams, ActorError> {
+    let payload: TokensReceivedParams = deserialize(raw, "receiver hook payload")?;
+    // Payload to address must match receiving actor.
+    if payload.to != my_id {
+        return Err(actor_error!(
+            illegal_argument,
+            "token receiver expected to {}, was {}",
+            my_id,
+            payload.to
+        ));
+    }
+    Ok(payload)
+}
+
+// Deserializes and validates a serialized allocation.
+fn validate_alloc_req(
+    serialized: &RawBytes,
+    policy: &Policy,
+    curr_epoch: ChainEpoch,
+    datacap_amount: &DataCap,
+) -> Result<AllocationRequests, ActorError> {
+    let reqs: AllocationRequests = deserialize(serialized, "allocation requests")?;
+
+    let mut allocation_total = StoragePower::zero();
+    for req in &reqs.requests {
+        allocation_total += StoragePower::from(req.size.0);
+
+        // Size must be at least the policy minimum.
+        if StoragePower::from(req.size.0) < policy.minimum_verified_allocation_size {
+            return Err(actor_error!(
+                illegal_argument,
+                "allocation size {} below minimum {}",
+                req.size.0,
+                policy.minimum_verified_allocation_size
+            ));
+        }
+
+        // Term must be at least the policy minimum.
+        if req.term_min < policy.minimum_verified_allocation_term {
+            return Err(actor_error!(
+                illegal_argument,
+                "allocation term min {} below limit {}",
+                req.term_min,
+                policy.minimum_verified_allocation_term
+            ));
+        }
+        // Term cannot exceed the policy maximum.
+        if req.term_max > policy.maximum_verified_allocation_term {
+            return Err(actor_error!(
+                illegal_argument,
+                "allocation term max {} above limit {}",
+                req.term_max,
+                policy.maximum_verified_allocation_term
+            ));
+        }
+        // Term range must be non-empty.
+        if req.term_min > req.term_max {
+            return Err(actor_error!(
+                illegal_argument,
+                "allocation term min {} exceeds term max {}",
+                req.term_min,
+                req.term_max
+            ));
+        }
+
+        // Allocation must expire soon enough.
```

Comment:
Since we're checking for non empty range maybe check for non empty expiration period `req.expiration > curr_epoch`